test/ng/compileSpec.js

```diff
@@ -12111,6 +12111,620 @@ describe('$compile', function() {
+            element = $compile('<' + tag + ' ng-prop-src="testUrl"></' + tag + '>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsUrl('untrusted:foo()');
+            $rootScope.$digest();
+            expect(element.prop('src')).toEqual('untrusted:foo()');
+
+            // As a RESOURCE URL
+            element = $compile('<' + tag + ' ng-prop-src="testUrl"></' + tag + '>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsResourceUrl('untrusted:foo()');
+            $rootScope.$digest();
+            expect(element.prop('src')).toEqual('untrusted:foo()');
+          }));
+        });
+      }
+    });
+
+    // Support: IE 9 only
+    // IE 9 rejects the `source` / `track` tags with
+    // "Unable to get value of the property 'childNodes': object is null or undefined"
+    if (msie !== 9) {
+      ['source', 'track'].forEach(function(tag) {
+        describe(tag + '[src]', function() {
+          it('should NOT require trusted values for whitelisted URIs', inject(function($rootScope, $compile) {
+            element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+            $rootScope.testUrl = 'http://example.com/image.mp4'; // `http` is whitelisted
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('http://example.com/image.mp4');
+          }));
+
+          it('should accept trusted values', inject(function($rootScope, $compile, $sce) {
+            // As a MEDIA_URL URL
+            element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsMediaUrl('javascript:foo()');
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+
+            // As a URL
+            element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsUrl('javascript:foo()');
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+
+            // As a RESOURCE URL
+            element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsResourceUrl('javascript:foo()');
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+          }));
+        });
+      });
+    }
+
+    describe('img[src] sanitization', function() {
+
+      it('should accept trusted values', inject(function($rootScope, $compile, $sce) {
+        element = $compile('<img ng-prop-src="testUrl"></img>')($rootScope);
+        // Some browsers complain if you try to write `javascript:` into an `img[src]`
+        // So for the test use something different
+        $rootScope.testUrl = $sce.trustAsMediaUrl('someuntrustedthing:foo();');
```

Comment:
Note that for the `ng-attr-` version of this test this string was `someUnstrustedThing:foo()`. But it seems like for properties the browser (at least some) change the protocol to lowercase when assigned to the property, so the `expect` below failed (which also has the mixed case for the attribute version). I changed both this line and the `expect` to be all lower case to avoid this since I don't think it matters here...